api/dataprocessor.go

```diff
@@ -188,6 +188,8 @@ func (s *Server) getTargets(reqs []models.Req) ([]models.Series, error) {
 	if len(localReqs) > 0 {
 		wg.Add(1)
 		go func() {
+			// the only errors returned are from us catching panics, so we should treat them
+			// all as internalServerErrors
```

Comment:
as of #378 more and more http things are making their way into the dataprocessor.
I think the dataprocessor should be an internal library for data processing, of which the http api server is just a user.
I think by itself that's a good way to organise the code, but also in the future we'll have cli tools which will essentially be a cli interface instead of an http interface, which will necessicate something like this.

such a refactor can be made later in a separate PR (or now if you want) but I don't think we shouldn't speak about internalServerErrors here since those are http concepts. since this library is an internal library it seems obvious to me that errors are "internal" errors.